composer/utils/checkpoint.py

```diff
@@ -256,17 +276,39 @@ def _download_checkpoint(
     return composer_states_filepath, extracted_checkpoint_folder, extracted_rank_n
 
 
+def glob_filter(exclude_globs: List[List[str]]) -> Callable[[Dict], None]:
+    """Provides a function which deletes all subparts of a dictionary based on a list of paths."""
+
+    def filter_func(state_dict: Dict) -> None:
+        for exclude_glob in exclude_globs:
+            temp_dict = state_dict
+            for step in exclude_glob[:-1]:
+                temp_dict = temp_dict[step]
+            del temp_dict[exclude_glob[-1]]
```

Comment:
Would it be possible to use https://docs.python.org/3.8/library/fnmatch.html to evaluate globs? (Though matching against '**' may require some custom logic -- it looks like fnmatch doesn't support that :'( )

Reply:
I avoided this because we're indexing into keys, and I don't want to flatten the state_dict into a series of paths, match, and then go back and delete when recursing is a little simpler. Let me know if there's a particular reason you think this would be better?